src/mt.rs

```diff
@@ -303,17 +303,19 @@ impl Mt19937GenRand32 {
     #[inline]
     pub fn fill_bytes(&mut self, dest: &mut [u8]) {
         const CHUNK: usize = size_of::<u32>();
-        let mut left = dest;
-        while left.len() >= CHUNK {
-            let (next, remainder) = left.split_at_mut(CHUNK);
-            left = remainder;
+        let mut left = dest.chunks_exact_mut(CHUNK);
+
+        for next in &mut left {
             let chunk: [u8; CHUNK] = self.next_u32().to_le_bytes();
             next.copy_from_slice(&chunk);
         }
-        let n = left.len();
+
+        let remainder = left.into_remainder();
+        let n = remainder.len();
+
         if n > 0 {
```

Comment:
Sounds good! I'll be honest the godbolt stuff is over my head right now, but I'll add `memcpy` to the list of things to look up, including `#[inline]`! Clippy had some suggestions:
```rust
dest_chunks
    .into_remainder()
    .iter_mut()
    .zip(self.next_u32().to_le_bytes().iter())
    .for_each(|(cell, &byte)| {
        *cell = byte;
    });
```

Reply:
That looks pretty nice.

We won't be able to do this because `rand_mt` supports Rust 1.47.0, but in newer Rust, there is a by value iterator on the `[u8; 4]` returned from `u32::to_le_bytes`, which would let the code look like this:

```rust
dest_chunks
    .into_remainder()
    .iter_mut()
    .zip(self.next_u32().to_le_bytes().into_iter())
    .for_each(|(cell, byte)| {
        *cell = byte;
    });
```

(because the bytes are moved by value, we don't need to dereference them with the `&` pattern in the closure)

Reply:
Oh and to get you pointed in the right direction on `memcpy`, this is one of the foundational functions of the C runtime that copies bytes from one spot to another. 

When `memcpy` appears in the godbolt output like that, that is Rust telling LLVM to use its memcpy "intrinsic": https://llvm.org/docs/LangRef.html#llvm-memcpy-intrinsic

Intrinsics are kinda like native code that exposes the guts of the machine code LLVM generates. In Artichoke, you could consider `String.new(capacity: 100)` an intrinsic that exposes a byte buffer.